cy_canvas/containers/CyReact.js

```diff
@@ -82,10 +82,14 @@ const Cy = React.createClass({
     var node = evt.cyTarget;
     this._cxtdragStart = true;
     this._edgeFrom = node;
+    this._edgeType = node.hasClass("user_says")? "us2r": "r2us"
   },
 
   dragOverNode: function(evt) {
     var node = evt.cyTarget;
+    if (node.hasClass("user_says") ^ this._edgeType == "r2us") {
```

Comment:
From my understanding, the purpose of this line of code is to prevent:
1. user_says => user_says (first expression)
2. response => response (second expression)

which means we only allow
1. user_says => response
2. response => user_says

Is it possible we keep the local variable edgeType but also use another variable to remember the type of the source node to enhance the code readability? It'll probably look like the following:

if (source.hasClass("user_says") ^  target.hasClass("response")) OR
if (source.hasClass("response") ^  target.hasClass("user_says")) OR
To use a function like isIntentPair(source, target) to express the intention of this line of code explicitly and hide the logic behind it